config/environments/production.rb

```diff
@@ -40,10 +40,9 @@
 
   # Assume all access to the app is happening through a SSL-terminating reverse proxy.
   # Can be used together with config.force_ssl for Strict-Transport-Security and secure cookies.
-  # config.assume_ssl = true
-
+  config.assume_ssl = ENV["DISABLE_SSL"].blank?
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
-  config.force_ssl = true
+  config.force_ssl  = ENV["DISABLE_SSL"].blank?
```

Comment:
```suggestion
  config.force_ssl  = ENV.include?("FORCE_SSL")
```

I think the default assumption is that you're using a reverse proxy to terminate incoming HTTPS traffic.

Reply:
We've been back and forth on this one quite a bit, but as I was going back through prior discussions, found this comment, which IMO is the clearest solution offered to this:

https://github.com/maybe-finance/maybe/issues/308#issuecomment-2081522347

Reply:
> it's just that I've never seen an example docker-compose bind the network interface (in like, any self-hosted software I use or have tried).

@gantoine Fair enough. I'm not really going to say that `everyone does it so it's a good idea` is ever going to convince me, especially once you realize how much automated traffic scanning is going on nowadays. Exposing a web application directly is usually a bad idea considering that webservers like Puma, Webrick etc. are not well-equipped to serve static assets.

@zachgoll I think we should provide a `docker-compose.yml` in the repository and a `docker-compose.override.yml` example in the documentation. By default, Docker Compose uses both `docker-compose.yml` and `docker-compose.override.yml` when you invoke `docker compose up -d` so we can support two scenarios out of the box with no convoluted tinkering required:

1. Bring your own reverse proxy - in this case the self-hoster has just to use the `docker-compose.yml`  in the repository and configure the reverse proxy on their own
2. `I don't care just let me run this` - in this case, we would ask the self-hoster to additionally create `docker-compose.override.yml` in the repository which would set up the reverse proxy for them, so literally no work would be required. Just copy both files and run `docker compose up -d` and you're ready to go.

In both scenarios, binding the backend port to the loopback interface (`127.0.0.1`) only prevents external connections from accessing the backend directly on port 3000.

If someone **actually** wants to expose the backend directly, then they can use `docker-compose.override.yml` to override our default binding.